### PR TITLE
Updated several vector db docs to fix errors and to expand on instructions for successful outcomes

### PR DESCRIPTION
**Pull request by mkheck**
Fixed several documentation bugs across the following vector DBs:
- Postgres/PGvector
- Neo4j
- Azure AI Search

Added required properties/code samples to successfully implement Neo4j as a VectorStore. These were missing.

Other docs fixes as appropriate.

### CODE REVIEW

File: models/spring-ai-azure-openai/README_AZURE_OPENAI.md
Comment (tzolov) [inline models/spring-ai-azure-openai/README_AZURE_OPENAI.md:1]: This README doesn't exist anymore. 
All azure docs are under [254b8632cd0a4235fd2d519ed2de51a70f6da841](https://github.com/spring-projects/spring-ai/blob/main/spring-ai-docs/src/main/antora/modules/ROOT/pages/api/clients/azure-openai-chat.adoc)

File: spring-ai-docs/src/main/antora/modules/ROOT/pages/api/vectordbs/neo4j.adoc [hunk 1]
```diff
@@ -15,16 +15,16 @@ link:https://neo4j.com/docs/cypher-manual/current/indexes-for-vector-search/[Neo
...
 c. link:https://neo4j.com/cloud/aura-free/[Neo4j Aura]
 d. link:https://neo4j.com/deployment-center/[Neo4j Server] instance
 
 == Configuration
 
-To connect to Neo4j and use the `Neo4jVectorStore`, you need to provide (e.g. via `application.properties`) configurations for your instance.
+To connect to Neo4j and use the `Neo4jVectorStore`, you need to provide (e.g. via environment variables) access details for your instance.
 
-Additionally, you'll need to provide your OpenAI API Key. Set it as an environment variable like so:
+Additionally, you will need to provide your OpenAI API Key. Set it as an environment variable like sos:
```
Comment (tzolov) [inline spring-ai-docs/src/main/antora/modules/ROOT/pages/api/vectordbs/neo4j.adoc:27]: I guess it is should be `so:` instead of `sos:`?

**Review by tzolov — CHANGES_REQUESTED** [review]
Thanks @mkheck, 

Can you please check my comments and rebase the PR. 
Cheers,

File: spring-ai-docs/src/main/antora/modules/ROOT/pages/api/vectordbs/neo4j.adoc (hunk 1 above)
Comment (mkheck) [reply spring-ai-docs/src/main/antora/modules/ROOT/pages/api/vectordbs/neo4j.adoc]: Definitely. 😅

**Comment by mkheck** [comment]
Good morning, @tzolov, please let me know if you need anything else. Cheers!

**Review by tzolov — APPROVED** [review]

**Comment by tzolov** [comment]
Merged! Thanks @mkheck

**Pull request merged**